### DISCUSSION TITLE
Change inverter working mode from a micropython controller

### DISCUSSION THREAD

**Discussion by mdughera** [mletenay/home-assistant-goodwe-inverter#176] [Ideas]
Hi experts, 

I know that this is probably OT, but it seems that this is the only place where there is knowledge on how to control Goodwe ET inverters via modbus/UDP.
I have been able to write a micropython program that connects to the inverter and reads runtime data. The program is running on a ESP8266 board. 
I am stuck halfway in being able to switch from General mode to Forced off-grid. 
Anybody can tell or guide me where to find information on the register index to be set? 

What I have discovered so far:
0xB798 (working mode: it works)
0xB997 (offline mode, obtained by reading the file et.py, doesn't work)
0xB0C0 (cold start, same as above)
0xB0C4 (backup_supply, same as above)

Anybody can help even though this is a topic not strictly related to HA? 

TIA

**Comment by mdughera** [comment]
Hi again,

I have been able to turn on and off Forced grid mode from a, esp8266 board. 
The script is in uPython.
The files in this project and the Goodwe Python library were essential to let me understand how to do it.

**Comment by mletenay** [comment]
You have meanwhile found it, but just for record, everything can be figured out from https://github.com/marcelblijleven/goodwe/blob/master/goodwe/et.py
